janus_server/src/datastore.rs

```diff
@@ -1163,6 +1163,44 @@ impl Transaction<'_> {
         }))
     }
 
+    /// Returns the number of rows in `aggregate_share_jobs` whose batch interval contains
+    /// `batch_unit_interval`.
+    #[tracing::instrument(skip(self), err)]
+    pub(crate) async fn get_aggregate_share_job_count_by_batch_unit(
```

Comment:
nit: this interval isn't necessarily a batch unit, so the method is somewhat confusingly named. Maybe `..._by_interval` instead of `..._by_batch_unit`?